### DISCUSSION TITLE
No way to remove authStore.onChange() handler

### DISCUSSION THREAD

**Discussion by atheros** [pocketbase/pocketbase#5350] [Q&A] [answered]
Hi, pb.authStore is the global state for auth, and I tried to listen for changes locally using .onChange().

However very soon I've noticed there is no way remove this listener from auth store, so if I use this method with components, I'll introduce a memory leak.

I know I can work around it by having a separate state that would listen for that change in a single place in my app, however that seem to be redundant.

I wouldn't mind a handy method for either removing this listener or for onChange() to return a handle to can somehow be freed.

Or am I missing something?

**Comment by ganigeorgiev** [comment, marked as answer]
I assume you are using the JS SDK. Then in this case you can call the returned "destroy" function, something like:

```js
const removeListener = pb.authStore.onChange((token, model) => {
    console.log('New store data 1:', token, model)
});
removeListener()
```

See the examples in https://github.com/pocketbase/js-sdk?tab=readme-ov-file#auth-store.

**Reply by atheros** [reply]
Oh, yeah, that works, thanks!
